app/src/main/java/com/cmput301f20t21/bookfriends/ui/profile/ProfileEditDialog.java

```diff
@@ -56,6 +71,28 @@ public void onClick(View v) {
                     listener.onEdit(inputEmail, inputPhone);
                     getDialog().dismiss();
                 }
+                //update email authentication
+                FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+                user.updateEmail(inputEmail)
+                        .addOnCompleteListener(new OnCompleteListener<Void>() {
+                            @Override
+                            public void onComplete(@NonNull Task<Void> task) {
+                                if (task.isSuccessful()) {
+                                    Log.d(TAG, "User email address updated.");
+                                }
+                            }
+                        });
+                //update field "email" and "phone"
+                User firebaseUser = AuthService.getInstance().getCurrentUser();
+                if (firebaseUser != null) {
+                    String userId = firebaseUser.getUid();
+                    FirebaseFirestore.getInstance().collection("users").document(userId)
+                            .update(
+                                    "email", inputEmail,
+                                    "phone", inputPhone
+                            );
```

Comment:
suggestion: after we moved the whole firebase-related section (line 74 ~ 93) to view model, I think we should still extract this part to UserService as a new method called `updateFirestoreUser`, which takes a HashMap<String, Object> and do the update call there.